### ISSUE TITLE
Type is not checked when spreading a typed object into another typed object

### ISSUE THREAD

**Issue by vsheyanov** [microsoft/TypeScript#58835]
### 🔎 Search Terms

"spread object", "spread typed object"

### 🕗 Version & Regression Information

Tested with TS:
- 5.0.4
- 5.4.5

### ⏯ Playground Link

https://www.typescriptlang.org/play/?#code/MYewdgzgLgBAbgLhgbxlAptJ0BOBLMAcxgF8YBeFAKBjUyiQHIATRgGipIG4qqB6PjAgALEAFcANsxgAjdDACGYGAFEcOEDiqhIsAqHXpgDFDDzNsUfEVIVqtAHRO4HWuaasO3XgKGiA7hAwACoAyjDo6pra4NAwBjhGJqjuQlYExGSUyDR0WDAs7LmphV5cQA

### 💻 Code

```ts
const v: { test: string } = {
  test: 'd',
};

// should be an Error
const incorrect: { id: string } = {
  ...v,
  id: 'd',
};

// shows TS error
const correct: { id: string } = {
  test: 'd',
  id: 'd',
};
```

### 🙁 Actual behavior

It allows to spread anything inside the variable `const incorrect`. You can even spread `...[],` inside `incorrect` variable and it won't throw an error.

### 🙂 Expected behavior

If variable `v` is typed, `incorrect` case must throw an error.

**Comment by MartinJohns** [comment]
Another duplicate of #39998. Type checking is working, just excess property checks not.

**Comment by vsheyanov** [comment]
@MartinJohns thanks for adding a duplicate. I didn't find that one myself.
I wonder if that issue covers also this case:
```
// should be an Error
const incorrect: { id: string } = {
  ...[],
  id: 'd',
};
```

**Comment by MartinJohns** [comment]
#52060 / #9726.
